tests/integration/renderer/test_persistence.py

```diff
@@ -0,0 +1,353 @@
+from multiprocessing import Value
+import dash
+import dash_html_components as html
+import dash_table as dt
+
+from dash.dependencies import Input, Output
+
+
+def table_columns(names, **extra_props):
+    return [dict(
+        id='c{}'.format(i),
+        name=n,
+        renamable=True,
+        hideable=True,
+        **extra_props
+    ) for i, n in enumerate(names)]
+
+
+def simple_table(names=('a', 'b'), **props_override):
+    props = dict(
+        id='table',
+        columns=table_columns(names),
+        data=[{'c0': 0, 'c1': 1}, {'c0': 2, 'c1': 3}],
+        persistence=True
+    )
+    props.update(props_override)
+    return dt.DataTable(**props)
+
+
+def reloadable_app(**props_override):
+    app = dash.Dash(__name__)
+    app.persistence = Value('i', 1)
+
+    def layout():
+        return html.Div([
+            html.Div(id='out'),
+            simple_table(persistence=app.persistence.value, **props_override)
+        ])
+    app.layout = layout
+
+    @app.callback(
+        Output('out', 'children'),
+        [Input('table', 'columns'), Input('table', 'hidden_columns')]
+    )
+    def report_props(columns, hidden_columns):
+        return 'names: [{}]; hidden: [{}]'.format(
+            ', '.join([col['name'] for col in columns]),
+            ', '.join(hidden_columns or [])
+        )
+
+    return app
+
+
+def rename_and_hide(dash_duo, rename=0, new_name='x', hide=1):
```

Comment:
> do you think it makes sense to create a small set of APIs to handle common dash operations in the future?

Sure, but only after we find that we're reusing similar functionality in multiple places. So far this one seems pretty specialized to me.

> I would probably make the new_name more "newer" than a simple 'x'

Good call -> https://github.com/plotly/dash/pull/903/commits/5fa4d559b54f7c7e10209188f4f3673bed76d90e

> I also like to add sometimes randomness to avoid mistakes in tests

As long as the test outcome is completely deterministic (to the best of our abilities) I suppose, but my preference is single-use unusual but not really random values.